--- a/include/obstack.h        2020-09-19 09:09:17.840475049 +0200
+++ b/include/obstack.h        2020-09-19 09:07:05.658225337 +0200
@@ -412,6 +412,7 @@
    if (__o->next_free + 1 > __o->chunk_limit)				\
      _obstack_newchunk (__o, 1);					\
    *(__o->next_free)++ = (datum);					\
+   __o->next_free += sizeof(void *);			\
    (void) 0; })
 
 /* These assume that the obstack alignment is good enough for pointers or ints,
@@ -423,7 +424,7 @@
 ({ struct obstack *__o = (OBSTACK);					\
    if (__o->next_free + sizeof (void *) > __o->chunk_limit)		\
      _obstack_newchunk (__o, sizeof (void *));				\
-   *((void **)__o->next_free)++ = ((void *)datum);			\
+   *((void **)__o->next_free) = ((void *)datum);			\
    (void) 0; })
 
 # define obstack_int_grow(OBSTACK,datum)				\
